protocols/messages.go

```diff
@@ -72,3 +95,16 @@ func (se *SideEffects) Merge(other SideEffects) {
 	se.TransactionsToSubmit = append(se.TransactionsToSubmit, other.TransactionsToSubmit...)
 
 }
+
+type comparable[T any] interface {
```

Comment:
I just merged in https://github.com/statechannels/go-nitro/pull/383 so I think most of the `Equal` work in here can be removed 🎉